### PR TITLE
feat: Iterate over all plays in a playbook

### PR DESCRIPTION
**Pull request by m-horky**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

- Updates logging of the playbook verifier
- Corrects type hints of one method
- Ensures that all plays of a playbook have been iterated

### CODE REVIEW

File: insights/client/apps/ansible/playbook_verifier/__init__.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def exclude_dynamic_elements(snippet):
     """
     if 'insights_signature_exclude' not in snippet.get('vars', {}):
         raise PlaybookVerificationError(
-            "The playbook snippet does not have the key 'vars/insights_signature_exclude', "
+            "Play does not have the key 'vars/insights_signature_exclude', "
```
Comment (ptoscano) [inline insights/client/apps/ansible/playbook_verifier/__init__.py:171]: s/Play/The play/ as editor's notes; the same applies to all the changes in this commit, and also affects the tests

**Review by ptoscano — COMMENTED** [review]
generally this PR makes sense, thanks

would it be possible to add tests for the new changes? ie a playbook with two or more plays, the first play properly signed, and any of the following plays with an invalid signature; a variation to that would be a revoked play, rather than one with an invalid signature

File: insights/client/apps/ansible/playbook_verifier/__init__.py (hunk 1 above)
Comment (ptoscano) [reply insights/client/apps/ansible/playbook_verifier/__init__.py:171]: hm was this done? it seems not

Comment (m-horky) [reply insights/client/apps/ansible/playbook_verifier/__init__.py:171]: It was, it seems GitHub didn't mark it as such though.

**Review by ptoscano — APPROVED** [review]
LGTM, thanks!

@xiangce would you please take a look? thanks in advance!

**Comment by xiangce** [comment]
Looks good to me

**Review by xiangce — APPROVED** [review]

**Pull request merged**